### PR TITLE
[Bugfix] show CG with star mark after table dropped

### LINKED ISSUE #1539: Create a table after drop, the colocate group information is not deleted
### Steps to reproduce the behavior (Required)
1. CREATE TABLE t_1 (
                                    `id` int(18),
                                    `value` varchar(20)
                                )
                                ENGINE=OLAP
                                DUPLICATE KEY(`id`)
                                COMMENT "OLAP"
                                DISTRIBUTED BY HASH(`id`) BUCKETS 3
                                PROPERTIES (
                                    "replication_num" = "1",
                                    "storage_format" = "v2",
                                    "colocate_with" = "group1"
                                );
2. DROP TABLE t_1;
3. CREATE TABLE t_1 (
                                    `id` int(18),
                                    `value` varchar(20)
                                )
                                ENGINE=OLAP
                                DUPLICATE KEY(`id`)
                                COMMENT "OLAP"
                                DISTRIBUTED BY HASH(`id`) BUCKETS 3
                                PROPERTIES (
                                    "replication_num" = "1",
                                    "storage_format" = "v2",
                                    "colocate_with" = "group1"
                                );
4. SHOW PROC '/colocation_group';

### Expected behavior (Required)
there is 1 tableID in group1
### Real behavior (Required)
there are 2 tableIDs in group1
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.0

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #1539

## Problem Summary(Required) ：

Add a marker at the table that has just dropped when executing `SHOW PROC '/colocation_group';`

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|18|18|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/catalog/ColocateTableIndexTest.java [hunk 1]
```diff
@@ -44,4 +55,115 @@ public void testGroupId() {
         balancingGroups.remove(groupId2);
         Assert.assertTrue(balancingGroups.isEmpty());
     }
+
+    @Test
+    public void testDropTable() throws Exception {
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/catalog/ColocateTableIndexTest.java:60]: Could you create a new test file and make it under our license header?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|18|18|100.00%|

**Review by gengjun-git — APPROVED** [review]

**Review by chaoyli — APPROVED** [review]

**Comment by chaoyli** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|18|18|100.00%|

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.3 branch-2.2 branch-2.1 branch-2.0
Lo siento, ¡tenemos demasiadas ramas!

**Pull request merged**